### DISCUSSION TITLE
How to get a list of all available SSO providers

### DISCUSSION THREAD

**Discussion by prayag17** [9p4/jellyfin-plugin-sso#216] [Q&A]
Hello, I am working on a Jellyfin Desktop client for Jellyfin called Blink and I was looking for a way to implement this plugin into my client natively. Is there a way to fetch all the available SSO config through an api endpoint?

**Comment by 9p4** [comment]
There is currently no way to get the list of available providers without authentication. If the API is called with auth, then you can get a list of providers. If you want to submit a PR to add that endpoint in, I would accept it.

**Reply by prayag17** [reply]
I was looking at the codebase when I found this https://github.com/9p4/jellyfin-plugin-sso/blob/main/SSO-Auth/Api/SSOController.cs#L369 tho this only returns the auth provider names, it can be used as a base for adding a new api endpoint like `/GetProviders` which would return all available providers which can be consumed by other clients. I will look into making a PR when I get the time to do so...
